sdk/core/azure-core/test/ut/policy.cpp

```diff
@@ -44,6 +44,28 @@ struct TestRetryPolicySharedState : public Azure::Core::Http::Policies::HttpPoli
   }
 };
 
+struct TestContextTreeIntegrity : public Azure::Core::Http::Policies::HttpPolicy
+{
+  std::unique_ptr<HttpPolicy> Clone() const override
+  {
+    return std::make_unique<TestContextTreeIntegrity>(*this);
+  }
+
+  std::unique_ptr<Azure::Core::Http::RawResponse> Send(
+      Azure::Core::Http::Request& request,
+      Azure::Core::Http::Policies::NextHttpPolicy nextHttpPolicy,
+      Azure::Core::Context const& ctx) const override
+  {
+    EXPECT_TRUE(ctx.HasKey("TheKey"));
+    if (ctx.HasKey("TheKey"))
```

Comment:
Since we assert `EXPECT_TRUE(ctx.HasKey("TheKey"));`, remove the `if` check on line 60?

Reply:
Why would the test execution continue? Wouldn't the assert failure on line 59 cause the test to fail?

Reply:
> Why would the test execution continue? 

Usually that's how all test frameworks behave. We want all test to run and at the end display which passed and which failed.
If we stop as soon as there is a failed test, we don't know if that's the only fail or not.

Reply:
> We want all test to run and at the end display which passed and which failed.

Right, that makes sense.

> If we stop as soon as there is a failed test, we don't know if that's the only fail or not.

I mean this particular test method (not all other tests). If the assert on line 59 fails, doesn't the method exit right there, and line 60 never runs (and then the framework continues running other tests)? That is why I am asking why we need the if check on line 60 at all, it will never be false.